### PR TITLE
Moved packages we use solely in doc examples and tests to devDependencies

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
Phase 1 of work for https://github.com/cerner/terra-ui/issues/165

### CODE REVIEW

File: packages/terra-aggregator/package.json [hunk 1]
```diff
@@ -26,8 +26,9 @@
     "react-dom": "^16.8.5"
   },
   "dependencies": {
-    "classnames": "^2.2.5",
```
Comment (bjankord) [inline packages/terra-aggregator/package.json]: classnames is not used in terra-aggregator so I've removed it altogether from the package.json file.

File: packages/terra-abstract-modal/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,11 @@ Changelog
 Unreleased
 ----------
 
+Unreleased
```
Comment (emilyrohrbough) [inline packages/terra-abstract-modal/CHANGELOG.md]: Whoops added Unreleased

File: packages/terra-slide-panel-manager/package.json [hunk 3]
```diff
@@ -29,14 +29,16 @@
   "dependencies": {
     "classnames": "^2.2.5",
     "prop-types": "^15.5.8",
+    "terra-disclosure-manager": "^4.8.0",
```
Comment (emilyrohrbough) [inline packages/terra-slide-panel-manager/package.json]: Should the disclosure manager be a peer dep?

File: packages/terra-abstract-modal/CHANGELOG.md (hunk 2 above)
Comment (bjankord) [reply packages/terra-abstract-modal/CHANGELOG.md]: De-duped in this commit: https://github.com/cerner/terra-framework/pull/626/commits/b2ba9614b3f3761e7459d2f7efe6d5a9cf3d1d87

File: packages/terra-slide-panel-manager/package.json (hunk 3 above)
Comment (bjankord) [reply packages/terra-slide-panel-manager/package.json]: Not sure, I'll defer to @tbiethman on this

Comment (tbiethman) [reply packages/terra-slide-panel-manager/package.json]: It should be, but moving it to peer now won't bepassive. I'll get a PR out to make it a peer dependency (and the one in terra-modal-manager) that we can time with another major release.

Comment (tbiethman) [reply packages/terra-slide-panel-manager/package.json]: Basically any component that provides Context-based APIs should be a peer dependency in non-application component packages. I want to warn on this at build-time with a webpack plugin or something; I might work on that over my sabbatical.

**Pull request merged**